numpy/core/src/multiarray/item_selection.c

```diff
@@ -2297,10 +2457,17 @@ PyArray_CountNonzero(PyArrayObject *self)
 
     /* Special low-overhead version specific to the boolean type */
     dtype = PyArray_DESCR(self);
-    if (dtype->type_num == NPY_BOOL) {
+
+    if (dtype->type_num >= NPY_INT16 && dtype->type_num <= NPY_UINT64) {
+        return count_nonzero_int(PyArray_NDIM(self), (void *) PyArray_DATA(self),
+                        PyArray_DIMS(self), PyArray_STRIDES(self), dtype->type_num);
+    }
+
+    if (dtype->type_num == NPY_BOOL || dtype->type_num == NPY_INT8 || dtype->type_num == NPY_UINT8) {
         return count_boolean_trues(PyArray_NDIM(self), PyArray_DATA(self),
                         PyArray_DIMS(self), PyArray_STRIDES(self));
     }
+
```

Comment:
would it be better to pass the element size instead? you can also handle u8/bool inside 
`count_nonzero_int`.

```suggestion
    switch(dtype->kind) {
        case 'u':
        case 'i':
        case 'b':
            if (dtype->elsize > 8) {
                break;
            }
            return count_nonzero_int(
                PyArray_NDIM(self), PyArray_DATA(self), PyArray_DIMS(self),
                PyArray_STRIDES(self), dtype->elsize
            );
    }
```

Reply:
```count_nonzero_int``` and ```count_boolean_true``` are a bit a different in the way they handle the optimization. The latter also includes an optimization for non simd cases too. So, I thought to keep these two types separated. Perhaps, we should keep it as it is?

Reply:
> count_nonzero_int and count_boolean_true are a bit a different in the way they handle the optimization

what I'm suggesting is using `count_nonzero_int` as a router for element sizes.

for now, let's fix the CI errors by passing the element size rather than the data type because I think the issue in `if (dtype->type_num >= NPY_INT16 && dtype->type_num <= NPY_UINT64) {`.

```suggestion
switch(dtype->kind) {
        case 'u':
        case 'i':
        case 'b':
           if (dtype->elsize > 8) {
                break;
           }
           if (dtype->elsize == 1) {
                return count_boolean_trues(PyArray_NDIM(self), PyArray_DATA(self),
                        PyArray_DIMS(self), PyArray_STRIDES(self));
           }
           return count_nonzero_int(
                PyArray_NDIM(self), PyArray_DATA(self), PyArray_DIMS(self),
                PyArray_STRIDES(self), dtype->elsize
            );
    }
````
then use `dtype->elsize`(size in bytes) inside `count_nonzero_int()` instead of data types.